share/utils.py

```diff
@@ -2,7 +2,16 @@
 # or more contributor license agreements. Licensed under the Elastic License 2.0;
 # you may not use this file except in compliance with the Elastic License 2.0.
 import hashlib
+import os
+
+from share.version import version
 
 
 def get_hex_prefix(src: str) -> str:
     return hashlib.sha3_384(src.encode("utf-8")).hexdigest()
+
+
+def create_user_agent() -> str:
+    """Creates the 'User-Agent' header given the library name and version"""
+    # AWS_EXECUTION_ENV – The runtime identifier, prefixed by AWS_Lambda_ (for example, AWS_Lambda_Python3.9).
+    return f"ElasticServerlessForwarder/{version} ({os.getenv('AWS_EXECUTION_ENV')})"
```

Comment:
nit: best to assume (even though this may never be the case), that we are not always running on AWS. 

we could populate the header with information available regardless of environment e.g. (`sys.version`), and conditionally add additional environment specific data where available.

Reply:
more then "nit" :)

if it's in share it should have no reference to the cloud environment :)

we could pass the cloud specific data as parameter